domain/machine/state/state.go

```diff
@@ -411,6 +413,120 @@ WHERE st.machine_uuid = $machineUUID.uuid;
 	}, nil
 }
 
+// SetRunningAgentBinaryVersion sets the running agent binary version for the
+// provided machine uuid. Any previously set values for this machine uuid will
+// be overwritten by this call.
+//
+// The following errors can be expected:
+// - [machineerrors.MachineNotFound] if the machine does not exist.
+// - [coreerrors.NotSupported] if the architecture is not known to the database.
+func (st *State) SetRunningAgentBinaryVersion(
+	ctx context.Context,
+	machineUUID string,
+	version coreagentbinary.Version,
+) error {
+	db, err := st.DB()
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	type ArchitectureMap struct {
+		ID   int    `db:"id"`
+		Name string `db:"name"`
+	}
+	archMap := ArchitectureMap{Name: version.Arch}
+
+	archMapStmt, err := st.Prepare(`
+SELECT id AS &ArchitectureMap.id FROM architecture WHERE name = $ArchitectureMap.name
+`, archMap)
+	if err != nil {
+		return internalerrors.Capture(err)
+	}
+
+	type MachineAgentVersion struct {
+		MachineUUID    string `db:"machine_uuid"`
+		Version        string `db:"version"`
+		ArchitectureID int    `db:"architecture_id"`
+	}
+	machineAgentVersion := MachineAgentVersion{
+		MachineUUID: machineUUID,
+		Version:     version.Number.String(),
+	}
+
+	upsertRunningVersionStmt, err := st.Prepare(`
+INSERT INTO machine_agent_version (*) VALUES ($MachineAgentVersion.*)
```

Comment:
$machineAgentVersion.*, it should be the var but not the type.
Or both are ok, not sure.
If the test passes, I guess you can use both?

Reply:
Yeah as written is correct. We work on types in sqlair.